lib/Doctrine/DBAL/Types/ObjectType.php

```diff
@@ -20,15 +20,15 @@ class ObjectType extends Type
      */
     public function getSQLDeclaration(array $fieldDeclaration, AbstractPlatform $platform)
     {
-        return $platform->getClobTypeDeclarationSQL($fieldDeclaration);
+        return $platform->getObjectTypeDeclarationSQL($fieldDeclaration);
```

Comment:
Instead of introducing new Platform methods, could the object type use BLOB column instead? Otherwise, even if they me binary-safe on some platforms, CLOB fields are subject to character set conversion and therefore are unsafe to store binary data.